app/lib/oidc_client.rb

```diff
@@ -83,65 +79,45 @@ def tokens!(oidc_nonce: nil)
 
     {
       access_token: response[:access_token],
-      refresh_token: response[:refresh_token],
       id_token_jwt: id_token_jwt,
       id_token: id_token,
     }.compact
   rescue Rack::OAuth2::Client::Error
     raise OAuthFailure
   end
 
-  def userinfo(access_token:, refresh_token:)
+  def userinfo(access_token:)
     response = time_and_return "userinfo" do
       oauth_request(
         access_token: access_token,
-        refresh_token: refresh_token,
         method: :get,
         uri: userinfo_endpoint,
       )
     end
 
-    begin
-      response.merge(result: JSON.parse(response[:result].body))
-    rescue JSON::ParserError
-      raise OAuthFailure
-    end
+    JSON.parse(response.body)
+  rescue JSON::ParserError
+    raise OAuthFailure
   end
 
 private
 
-  class RefreshAndRetry < StandardError; end
+  class Retry < StandardError; end
 
-  class OnlyRetry < StandardError; end
-
-  REFRESH_AND_RETRY_STATUSES = [401, 403].freeze
```

Comment:
I'm a little unclear what happens in a 401 and 403 case now.
previously we would have retried... once (?), then hit `MAX_OAUTH_RETRIES` and then raised `OAuthFailure`...

Presumably that just renderes an error page (is that the "something has gone wrong" page? or something more descriptive?), or do we just boot a user back to the login form?

So now for 401 and 403 we just jump straight to those pages, right?
No retries because those are definative "we see you and you're not allowed further" statuses.

Do we think any error pages we might serve in response to these are descriptive enough?
I can't quite see how this'd look from a user's perspective if they ended up on this branch of code?

Reply:
This will be returned as a 401 from account-api to the consuming app.  Since the retry would have failed anyway (as there's no refresh token to use to get a new access token), there should be no change in response to other apps.